### DISCUSSION TITLE
Regarding Number of shots in PEC

### DISCUSSION THREAD

**Discussion by Julie-Maria-Raju** [unitaryfund/mitiq#2583] [Q&A]
Hallo,
I was trying to apply pec for a classification tasks and used execute_with_pec with source code having 1 shots. Why is it only 1 shots? I tried it with 1024 shots and got little better results than one shots comparing the accuracy. Have anybody tried pec for classification task? Please let me know.

**Comment by natestemen** [comment]
Thanks for the question, and cool to hear you're trying to use PEC!

> Why is it only 1 shots?

I'll need a little more context here as to what you mean. Are you basing your code off an example that uses a single shot? It sounds like you're saying we have something hard coded to take a single shot for PEC circuits, but I cannot find where, if that is the case.

> Have anybody tried pec for classification task?

Not that I'm aware of! What dataset are you working with/testing on? I'm curious to know more about the set up here.

**Reply by Julie-Maria-Raju** [reply]
Sorry it was my fault, I meant execute_with_noise uses one shot and why?

I am working on OrganAMNIST dataset for class of 11. My expectation values are changing but the classification accuracy doesn’t have an improvement. I expected a better result when I am using long computational time to just validate it with pec with trained weights.

**Reply by natestemen** [reply]
Ah I see what you mean. So the docstring indicates that the simulation is an exact one. I.e. it's computing $\text{tr}(O \rho)$ without any sampling.
https://github.com/unitaryfund/mitiq/blob/be6a76ce940033ddd58538f3fe9e989b8e536850/mitiq/interface/mitiq_qiskit/qiskit_utils.py#L86-L92

This is how you used to do an exact simulation with the `AerSimulator`, but I cannot find any documentation on it now. Is it working as expected, or are you seeing strange results?

If you want to use shots to emulate hardware more effectively there is also the [`execute_with_shots_and_noise`](https://github.com/unitaryfund/mitiq/blob/be6a76ce940033ddd58538f3fe9e989b8e536850/mitiq/interface/mitiq_qiskit/qiskit_utils.py#L128) function in the same file.

**Reply by Julie-Maria-Raju** [reply]
Yes , when I tried it with one shots it was giving less validation accuracy like 18 and when i used 1024 it is giving 56.5.
I also have doubt , I am now applying 500 samples for PEC and by increasing it does it improve the mitigation?

**Reply by natestemen** [reply]
Thanks for raising this. I've asked a question over on the qiskit-aer github page (https://github.com/Qiskit/qiskit-aer/discussions/2269). This might be a bug!

**Reply by natestemen** [reply]
Unfortunately haven't gotten a response from the Qiskit team, so I'm going to close this. If we hear back I will reopen with info.
